lib/engine/game/g_18_india/step/certificate_selection.rb

```diff
@@ -55,57 +58,75 @@ def players_visible?
             false
           end
 
+          def show_companies
+            true
+          end
+
+          def show_map
+            true
+          end
+
           def name
-            'Hand Selection'
+            'Initial Hand Selection'
           end
 
           def description
             "Select #{@cards_to_keep} Certificates for your starting hand"
           end
 
-          def finished?
-            @confirmed_selections == @game.players.size
+          def selection_note
+            [
+              "Made #{number_of_selections} of #{@cards_to_keep} selections.",
+              'Click on card to select or unselect it.',
+            ]
           end
 
-          def selections_completed?
-            number_of_selections == @cards_to_keep
+          def select_company(player, company)
+            return if selections_completed? && company.owner.nil? # prevent selecting more than allowed
+
+            # add or remove from choices
+            if company_selected?(company)
+              @choices[player].delete(company)
+              company.owner = nil
+            else
+              @choices[player] << company
+              @choices[player].sort!
```

Comment:
Does it matter whether this array is sorted?

Reply:
It's a bit nicer to see the companies in order on the player card.  You can see this in the screenshot above for player 5, which has the selections also listed on the player card...